File: php/features.php
<?php
declare(strict_types=1);

// IpIntelligence SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IpIntelligenceFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IpIntelligenceBaseFeature();
            case "test":
                return new IpIntelligenceTestFeature();
            default:
                return new IpIntelligenceBaseFeature();
        }
    }
}

File: py/features.py
# IpIntelligence SDK feature factory

from feature.base_feature import IpIntelligenceBaseFeature
from feature.test_feature import IpIntelligenceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpIntelligenceBaseFeature(),
        "test": lambda: IpIntelligenceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
